cl_hubeau/water_services/utils.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Convenience functions for water services indicators.
+"""
+
+import geopandas as gpd
+import pandas as pd
+from tqdm import tqdm
+
+from cl_hubeau.water_services.water_services_scraper import WaterServicesSession
+from cl_hubeau import _config
+from cl_hubeau.utils import get_departements_from_regions, get_regions, prepare_kwargs_loops
+from datetime import datetime
+
+def get_all_communes(**kwargs) -> gpd.GeoDataFrame:
+    """
+    Gets the services for every french city.
+
+    Parameters
+    ----------
+    **kwargs :
+        kwargs passed to WaterServicesSession.get_communes (hence mostly intended
+        for hub'eau API's arguments). Do not use `format` or `code_commune`
+        as they are set by the current function.
+
+    Returns
+    -------
+    results : gpd.GeoDataFrame
+        GeoDataFrame of water services indicators for every french city.
+
+    """
+    with WaterServicesSession() as session :
+        regions = get_regions(latest=True)
+        results = [
+            session.get_communes(
+                code_departement=get_departements_from_regions(reg), detail_service=True, format="geojson", **kwargs
+            )
+            for reg in tqdm(
+                regions,
+                desc="querying regions for communes",
+                leave=_config["TQDM_LEAVE"],
+                position=tqdm._get_free_pos(),
+            )
+        ]
+    results = [x[0].dropna(axis=1, how="all") for x in results if not x[0].empty]
+    results = gpd.pd.concat(results, ignore_index=True)
+    try:
+        results["code_commune_insee"]
+        results = results.drop_duplicates("code_commune_insee")
+    except KeyError:
+        pass
+    return results
+
+
+def get_all_services(**kwargs) -> pd.DataFrame:
+    """
+    Gets the services for every french city.
+
+    Parameters
+    ----------
+    **kwargs :
+        kwargs passed to WaterServicesSession.get_services (hence mostly intended
+        for hub'eau API's arguments). Do not use `format` or `code_commune`
+        as they are set by the current function.
+
+    Returns
+    -------
+    results : pd.DataFrame
+        DataFrame of water services indicators for every service.
+
+    """
+    with WaterServicesSession() as session:
+        regions = get_regions(latest=True)
+        results = []
+        for reg in tqdm(
+            regions, desc="Querying regions for services",
+            position=0
+        ):
+            for dep in tqdm(
+                get_departements_from_regions(reg),
+                desc="Querying departements from regions",
+                leave=False,
+                position=1
+            ):
+                result = session.get_services(code_departement=dep, **kwargs)
+                results.append(result)
```

Comment:
Same as I said for `get_all_communes` with another reason: `tqdm` doesn't work well when nested...